### PR TITLE
Make DOSGuard config loading more resilient

### PR DESCRIPTION
**Pull request by undertome**
Don't fail when the config is malformed

---
This change is [Reviewable](https://reviewable.io/reviews/cjcobb23/clio/103)

### CODE REVIEW

File: src/webserver/DOSGuard.h [hunk 1]
```diff
@@ -40,22 +46,26 @@ class DOSGuard
...
+            auto const transform = [](auto const& elem) {
+                return std::string(elem.as_string().c_str());
+            };
 
-        return T(
-            boost::transform_iterator(w.begin(), transform),
-            boost::transform_iterator(w.end(), transform));
+            return T(
+                boost::transform_iterator(w.begin(), transform),
+                boost::transform_iterator(w.end(), transform));
```
Comment (cjcobb23) [inline src/webserver/DOSGuard.h:63]: I know this is not actually part of this PR, but can we do this with the standard library instead of boost?

**Review by cjcobb23 — APPROVED** [review]
LGTM. Left a non blocking comment

File: src/webserver/DOSGuard.h (hunk 1 above)
Comment (undertome) [reply src/webserver/DOSGuard.h:63]: It depends on what you mean by that. There's no `transform_iterator` in the standard library.

Comment (undertome) [reply src/webserver/DOSGuard.h:63]: I don't think there's a concise way to do this without using boost.

**Pull request merged**